egrismore/Lab4.py

```diff
@@ -0,0 +1,27 @@
+__author__ = 'emilygrismore'
+"""checks requirements for and runs the magic 1089"""
+
+
+def flip(i):
+    """reverses the number"""
+    int(str(i)[::-1])
+    return i
+
+
+def decrease(i):
+    """checks to see if the #'s are in descending order"""
+    largest = 10
+    for new in str(i):
+        number = int(new)
+        if number >= largest:
+            return False
+        else:
+            largest = number
+    return True
+
+
+def trick(i):
+    """shows the magical 1089"""
+    difference = i - flip(i)
+    difference + flip(difference)
+    print(i)
```

Comment:
def trick(i):

    """shows the magical 1089"""

    difference = i - flip(i)

    final = difference + flip(difference)

    return final





from . import Lab4



def test_Lab4():

    for i in range(100, 999):

        if Lab4.decrease(i):

            assert Lab4.trick(i) == 1089


this is what I have right now


On Tue, Oct 13, 2015 at 12:32 PM, Emily Grismore <egrismore@anderson.edu>
wrote:

> That is exactly what I put for my assert line! What is a noop?
>
> On Tue, Oct 13, 2015 at 12:31 PM, Kyle Tarplee <notifications@github.com>
> wrote:
>
>> In egrismore/Lab4.py
>> <https://github.com/AndersonUniversity/CPSC4500test/pull/3#discussion_r41890619>
>> :
>>
>> > +    """checks to see if the #'s are in descending order"""
>> > +    largest = 10
>> > +    for new in str(i):
>> > +        number = int(new)
>> > +        if number >= largest:
>> > +            return False
>> > +        else:
>> > +            largest = number
>> > +    return True
>> > +
>> > +
>> > +def trick(i):
>> > +    """shows the magical 1089"""
>> > +    difference = i - flip(i)
>> > +    difference + flip(difference)
>> > +    print(i)
>>
>> Put tons of print() lines in your code and figure out what is happening.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/AndersonUniversity/CPSC4500test/pull/3/files#r41890619>
>> .
>>
>
>